mobile/lib/widgets/settings/networking_settings/external_network_preference.dart

```diff
@@ -0,0 +1,199 @@
+import 'dart:convert';
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/models/auth/auxilary_endpoint.model.dart';
+import 'package:immich_mobile/entities/store.entity.dart' as db_store;
+import 'package:immich_mobile/widgets/settings/networking_settings/endpoint_input.dart';
+
+class ExternalNetworkPreference extends HookConsumerWidget {
+  const ExternalNetworkPreference({super.key, required this.enabled});
+
+  final bool enabled;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final entries =
+        useState([AuxilaryEndpoint(url: '', status: AuxCheckStatus.unknown)]);
+    final canSave =
+        useState(entries.value.every((e) => e.status == AuxCheckStatus.valid));
+
+    saveEndpointList() {
+      canSave.value =
+          entries.value.every((e) => e.status == AuxCheckStatus.valid);
+
+      final endpointList = entries.value
+          .where((url) => url.status == AuxCheckStatus.valid)
+          .toList();
+
+      final jsonString = jsonEncode(endpointList);
+
+      db_store.Store.put(
+        db_store.StoreKey.externalEndpointList,
+        jsonString,
+      );
+    }
+
+    updateValidationStatus(String url, int index, AuxCheckStatus status) {
+      entries.value[index] =
+          entries.value[index].copyWith(url: url, status: status);
+
+      saveEndpointList();
+    }
+
+    handleReorder(int oldIndex, int newIndex) {
+      if (oldIndex < newIndex) {
+        newIndex -= 1;
+      }
+
+      final entry = entries.value.removeAt(oldIndex);
+      entries.value.insert(newIndex, entry);
+      entries.value = [...entries.value];
+
+      saveEndpointList();
+    }
+
+    handleDismiss(int index) {
+      entries.value = [...entries.value..removeAt(index)];
```

Comment:
`removeAt` returns the removed value. Shouldn't it be the opposite?

Reply:
So  `.removeAt()` returns the index, and the double dot `..removeAt()` returns the array

Reply:
https://dart.dev/language/operators#cascade-notation